### ISSUE TITLE
all previously uploaded global docs referenced instead of specific doc sources.

### ISSUE THREAD

**Issue by momuno** [microsoft/chat-copilot#671]
**Describe the bug**
After uploading multiple docs globally, when a query is sent in chat, instead of the specific docs that answer the question being references, ALL the global docs that have been uploaded are referenced.

**To Reproduce**
1. Start chat copilot via quickstart
2. Under documents tab, upload multiple different global docs.
3. Ask about information that should be found in only one or two of the docs.

**Expected behavior**
Only the docs that actually answer the question should be referenced.

**Screenshots**
![image](https://github.com/microsoft/chat-copilot/assets/64985898/8341f3ee-4405-4bdf-8d94-e190bbe06d60)

**Platform**
 - OS: Windows 11
 - IDE: VS Code
 - Source: https://github.com/microsoft/chat-copilot/commit/5ba7ed01068a2ffa4894331e5a6bd9a4420bcc17

**Comment by crickman** [comment]
This is a side-effect of importing the same doc to the same index multiple times.  As such, you recieved multiple valid results.

The citation lists all results included in the prompt as there's no way of knowing which the model prefered.  Each result is a specific node or chunk of a document...so a given document might have several cititions.

One might assert that since the file name is the same the citations might be reconcile3d...but the document identifiers are different.  There's no enforcment to prevent a user from uploading different documents with the same name and i'm not sure we'd want that.

In short, there's a real risk of creating a bug farm by getting too fancy here.
